numpy/linalg/linalg.py

```diff
@@ -2556,11 +2561,15 @@ def norm(x, ord=None, axis=None, keepdims=False):
             # special case for speedup
             s = (x.conj() * x).real
             return sqrt(add.reduce(s, axis=axis, keepdims=keepdims))
+        elif ord in ('f', 'fro'):
+            raise ValueError("Cannot compute Frobenius norm for vectors, "
+                             "vectors only support integer norm orders. For "
+                             "example for the L2 norm use ord=2 instead")
+        elif ord is 'nuc':
+            raise ValueError("Nuclear norm is not supported for vectors")
+        elif not isinstance(ord, numbers.Integral):
+            raise ValueError("Invalid norm order for vectors.")
```

Comment:
Is this additional condition necessary? Are there cases outside of the 'nuc' and 'fro' for which the `try-catch` you removed are on the path? Other (invalid) str kwargs?

If you can verify that doing so is sufficient for replacing the original `try-except` block that was removed, you could consider further consolidating the above check to:

```python
elif isinstance(ord, str):
    raise ValueError(f"Invalid norm order '{ord}' for vectors")
```

Either way, I think a check based on `numbers.Integral` should be avoided.

Reply:
Yeah I agree it is not optimal. I changed it to catch any string as you mentioned in case people have a typo in their `ord` string.